crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -47,6 +48,8 @@ where
     /// transaction without having to regenerate the signature or requiring access to the
     /// authenticator that produced it.
     generated_signatures: BTreeMap<Word, Vec<Felt>>,
+
+    source_manager: Arc<dyn SourceManagerSync>,
```

Comment:
Similar question as in the `miden-vm` PR: do we actually need the `source_manager` to be `Arc<dyn SourceManagerSync>` here? Maybe we could have it as a generic parameter? (or even just a concrete `DefaultSourceManger` for now?)